spring-java17/src/main/java/io/github/xanthic/cache/springjdk17/XanthicSpringCacheManager.java

```diff
@@ -60,16 +59,30 @@ public XanthicSpringCacheManager(Consumer<CacheApiSpec<Object, Object>> spec, @N
 	@Nullable
 	public Cache getCache(@NotNull String name) {
 		// Optimistic lock-free lookup to avoid contention: https://github.com/spring-projects/spring-framework/issues/30066
-		Cache optimistic = cacheMap.get(name);
-		if (optimistic != null || !dynamic)
-			return optimistic;
+		CacheWrapper optimistic = cacheMap.get(name);
+		if (optimistic != null)
+			return optimistic.cache();
+		if (!dynamic)
+			return null;
 
-		return this.cacheMap.computeIfAbsent(name, cacheName -> createCache(cacheName, this.spec));
+		return this.cacheMap.computeIfAbsent(name, cacheName -> new CacheWrapper(createCache(cacheName, this.spec))).cache();
 	}
 
 	@Override
 	public @NotNull Collection<String> getCacheNames() {
-		return this.cacheMap.keySet();
+		return Collections.unmodifiableSet(cacheMap.keySet());
+	}
+
+	@NotNull
+	public Set<String> getCustomCacheNames() {
```

Comment:
spring doesn't have this method, but we exposed it via `@Getter` - we can consider deprecating it / marking it as only visible for testing

Reply:
i think we can deprecate this, i can't think of a good use-case that isn't solved better in a different way without this method.